olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -235,6 +243,69 @@ IntrospectAppResult GetIntrospectAppResult(const rapidjson::Document& doc) {
   return result;
 }
 
+AuthorizeResult GetAuthorizeResult(rapidjson::Document& doc) {
+  AuthorizeResult result;
+
+  auto getPermition = [](const char* str) -> DecisionType {
+    if (strcmp(str, "allow") == 0) {
```

Comment:
strcmp is bad, please do not use it! Either use strncmp or create strings and compare strings.